### DISCUSSION TITLE
实现主题切换

### DISCUSSION THREAD

**Discussion by xcxiao7710** [ant-design/ant-design#51115] [General]
<img width="1677" alt="image" src="https://github.com/user-attachments/assets/a4925fda-e5a7-49cd-bc07-33d86f240c92">`import React from 'react';
import { Button, Switch } from 'antd';
import { CiSun, CiCloudMoon } from 'react-icons/ci';

import { setTheme } from '@/config/store/features/userInfoSlice';
import { useAppDispatch, useAppSelector } from '@/config/store/hooks';

const ThemeSwitcher: React.FC = () => {
  const dispatch = useAppDispatch();
  const theme = useAppSelector((state) => state.userInfo.theme);

  const onChange = () => {
    dispatch(setTheme(theme === 'dark' ? 'light' : 'dark'));
  };

  return (
    <span>
      <div onClick={onChange}>
        {theme === 'dark' ? <CiCloudMoon /> : <CiSun />}
      </div>
    </span>
  );
};

export default ThemeSwitcher;
``import React, { ReactNode, useEffect } from 'react';

import { ConfigProvider } from 'antd';
import zhCN from 'antd/locale/zh_CN';

import { setTheme } from '@/config/store/features/userInfoSlice';
import { useAppDispatch, useAppSelector } from '@/config/store/hooks';
import { darkTheme, defaultTheme } from '@/config/styles/themeConfig';

const getCurrentTheme = (): 'dark' | 'light' => {
  const hour = new Date().getHours();
  return hour >= 18 || hour < 6 ? 'dark' : 'light';
};

export const ThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const dispatch = useAppDispatch();
  const theme = useAppSelector((state) => state.userInfo.theme);

  useEffect(() => {
    if (!theme) {
      console.log('没有用户设置的主题，使用默认时间主题');
      const initialTheme = getCurrentTheme();
      dispatch(setTheme(initialTheme));
    }
  }, [dispatch, theme]);

  const appliedTheme = theme === 'dark' ? darkTheme : defaultTheme;

  return (
    <ConfigProvider theme={appliedTheme} locale={zhCN}>
      {children}
    </ConfigProvider>
  );
};

import {
  darkTheme as darkThemes,
  defaultTheme as defaultThemes,
} from '@ant-design/compatible';

export const darkTheme: any = {
  ...darkThemes,
  headerName: '商城管理系统',

  token: {
    colorPrimary: '#00B96B',
    colorBgMask: 'rgba(0, 0, 0, 0.1)',
    colorBgLayout: '#141414',
    colorBgContainer: '#141414',
  },
  components: {
    Layout: {
      headerBg: '#141414',
      siderBg: '#141414',
      triggerBg: '#141414',
    },

    Breadcrumb: {
      colorText: '#00B96B',
      colorPrimaryBorder: '#00B96B',
    },
  },
};

export const defaultTheme: any = {
  ...defaultThemes,
  // 系统名称，显示在页面标题或头部
  headerName: '商城管理系统',

  // 主题颜色和样式配置
  token: {
    colorPrimary: '#00B96B', // 主要操作颜色
    colorBgMask: 'rgba(0, 0, 0, 0.1)', // 遮罩背景色
    colorBgLayout: '#ffffff',
    colorBgContainer: '#ffffff',
  },
  // 组件特定的样式配置
  components: {
    Layout: {
      headerBg: '#ffffff',
      siderBg: '#ffffff',
      triggerBg: '#ffffff',
      triggerColor: '#00B96B',
    },
  },
};
`

**Comment by xcxiao7710** [comment]
在实现主题切换时 因为使用的ProCard 多次切换出现的css样式残留

**Reply by zhu0629** [reply]
最近才有的bug，我也发现了 尝试用上一个版本试试。

**Reply by zhu0629** [reply]
最近才有的bug，我也发现了 尝试用上一个版本试试。
